docs/tutorial.md

```diff
@@ -2,44 +2,50 @@
+writing stateful services.  Now, you can write a stateful service
+complete with
+persistent volumes, fault tolerance, and configuration management in
+about 100 lines of code.  This SDK is the product of Mesosphere's
+experience writing production stateful services such as [DC/OS Apache
+Kafka](https://docs.mesosphere.com/latest/usage/service-guides/kafka/), [DC/OS Apache Cassandra](https://docs.mesosphere.com/latest/usage/service-guides/cassandra/), and [DC/OS Apache
+HDFS](https://docs.mesosphere.com/latest/usage/service-guides/hdfs/).
+
+Read about the [features](#Features-at-a-glance) to learn more about the DC/OS
+Stateful Service SDK.
 
 # Introduction
 
-In this tutorial, we'll build a fictional but illustrative example
-service named `data-store`.  It is composed of `meta-data` nodes and
-`data` nodes.  This is meant to model a service with a small number of
-coordinating masters, and a large number of dependent workers.  Each
-task will run a dummy command that writes some data to a file, then
-sleeps.
+In this tutorial, we'll build an example
+service called `data-store`.  It is composed of a small number of
+`meta-data` nodes and a larger number of `data` nodes.  This structure
+is meant to model a service with a small number of coordinating master
+nodes and a large number of dependent agent nodes.  Each task runs a
+command that writes some data to a file, then sleeps.
 
 The goal state for this fictional `data-store` is to keep two
-`meta-data` and N `data` nodes running, where initially N = 3.
-`meta-data` nodes must come up before `data` nodes.  The service will
-scale by starting more `data` nodes.
+`meta-data` and N `data` nodes running, where, initially, N = 3.
+`meta-data` nodes must come up before `data` nodes.  The service
+scales by starting more `data` nodes.
 
-We'll include code snippets inline in this tutorial, and the final
+This tutorial includes code snippets. The final
 source code is available at
-http://github.com/mesosphere/sdk-reference-framework.
+[http://github.com/mesosphere/sdk-reference-framework](http://github.
+com/mesosphere/sdk-reference-framework).
 
 ## Architecture
 
 ![DC/OS Service Architecture](./architecture.png)
 
-A DC/OS Sercive is composed of **schedulers** and **executors**.  The
+A DC/OS service is composed of **schedulers** and **executors**.  The
 scheduler is run by Marathon, which supervises the scheduler,
-restarting it upon failure.  The executors are started and supervised
-by the scheduler.  If specified in the `ServiceSpecification`, the
+restarting it upon failure.  The scheduler, in turn, starts and
+supervises the executors.  If specified in the `ServiceSpecification`,
+the
 executors will contain a **persistent volume**, which is a folder or
-mount point on the agent that persists even after a task fails.
+mount point on the agent that persists even after a task fails. This
+allows a stateful service to recover with its data intact after node
```

Comment:
"after task failure or temporary node failure"

If it's a permanent failure, we can't recover the volume.